packages/pwa/app/commerce-api/hooks/useCustomerProductList.js

```diff
@@ -0,0 +1,325 @@
+             * @param {string} listId id of the list to update the item in
+             * @param {string} itemId the id of the item in the product list
+             */
+            async removeListItem(listId, itemId) {
+                const response = await api.shopperCustomers.deleteCustomerProductListItem(
+                    {
+                        parameters: {
+                            itemId,
+                            listId,
+                            customerId: customer.customerId
+                        }
+                    },
+                    true
+                )
+
+                if (isError(response)) {
+                    throw new Error(response)
+                }
+
+                return response
+            },
+
+            /**
+             * Fetches product lists for registered users or
+             * creates a new list if none exists, due to the api
+             * limitation, we can not get the list based on
+             * name/type, therefore it fetches all lists
+             * @returns product lists for registered users
+             */
+            async _getOrCreatelistByName(name, options) {
+                let response = await _super.getLists()
+
+                // Note: if list is empty, the API response doesn't
+                // contain the "data" key.
+                if (!response.data?.some((list) => list.name === name)) {
+                    const {id} = await _super.createList(name, options)
+                    response = await self.getList(id)
+                    return response
+                }
+
+                return response.data.find((list) => list.name === name)
+            },
+
+            /**
+             * Fetch list of product details from a product list.
+             * The maximum number of productIDs that can be requested are 24.
+             * @param {array} list product list
+             * @returns {Object[]} list of product details for requested productIds
+             */
+            async _getProductDetails(list) {
+                if (!list.customerProductListItems) {
+                    return
+                }
+
+                const ids = list.customerProductListItems.map((item) => item.productId)
+                const response = await api.shopperProducts.getProducts({
+                    parameters: {
+                        ids: ids.join(',')
+                    }
+                })
+
+                if (isError(response)) {
+                    throw new Error(response)
+                }
+
+                return response
+            },
+
+            _mergeProductDetailsIntoList(list, productDetails) {
+                list.customerProductListItems = list.customerProductListItems?.map((item) => {
+                    const detail = {
+                        ...productDetails.data.find((product) => product.id === item.productId)
+                    }
+                    return {
+                        ...detail,
```

Comment:
Lately I've been a stickler on naming things as what they are, can we not call the product a product and not detail?